### PR TITLE
Base phase abstraction and conversion of phase-lib to use it.

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: core/stages.js [hunk 1]
```diff
@@ -58,6 +58,7 @@ function readFile(filename, cb) {
 }
 
 function readFileRaw(filename, cb) {
+  console.log('reading', filename, 'raw');
```
Comment (dstockwell) [inline core/stages.js]: Kinda think we should remove all these, they just get in the way.

File: core/phase-lib.js [hunk 2]
```diff
@@ -2,55 +2,45 @@ var fs = require('fs');
...
-  return new stream.CoreStream(impl, 'toTag', undefined, types.string, types.string);
-}
 
-function dummy(options) {
-  function impl(items, cb) {
-    return items;
-  };
-  return new stream.CoreStream(impl, 'dummy', undefined, types.string, types.string);
-}
+register({name: 'log', input: types.string, output: types.string, arity: '1:1'},
```
Comment (dstockwell) [inline core/phase-lib.js]: We should make arity: '1:1' default.

**Comment by dstockwell** [comment]
lgtm

File: core/stages.js (hunk 1 above)
Comment (dstockwell) [inline core/stages.js]: https://github.com/dglazkov/tree-builder-builder/issues/105

File: core/phase-lib.js (hunk 2 above)
Comment (shans) [inline core/phase-lib.js]: It is.

**Pull request merged**